### PR TITLE
Add sourcehut to available urls' in Node class

### PR DESCRIPTION
**Pull request by TahlonBrahic**

### CODE REVIEW

File: src/nix_auto_follow/cli.py [hunk 1]
```diff
@@ -64,6 +64,11 @@ def get_url(self) -> str:
                 return f"github:{original['owner']}/{original['repo']}{rev_or_ref}"
             case "gitlab":
                 return f"gitlab:{original['owner']}/{original['repo']}{rev_or_ref}"
+            case "sourcehut":
+                return f"sourcehut:{original['owner']}/{original['repo']}{rev_or_ref}"
+            case "tarball":
+                # This is hella complicated
+                return ""
```
Comment (fzakaria) [inline src/nix_auto_follow/cli.py:71]: Why did you add it ?
Should we remove it now and let it fail.

File: src/nix_auto_follow/cli.py [hunk 2]
```diff
@@ -64,6 +64,11 @@ def get_url(self) -> str:
                 return f"github:{original['owner']}/{original['repo']}{rev_or_ref}"
             case "gitlab":
                 return f"gitlab:{original['owner']}/{original['repo']}{rev_or_ref}"
+            case "sourcehut":
```
Comment (fzakaria) [inline src/nix_auto_follow/cli.py:67]: If you look at the test file should be pretty straightforward to add a case for sourcehut; could you do that please?

**Review by fzakaria — COMMENTED** [review]
Thank you for your contribution!
Happy to add more.

File: src/nix_auto_follow/cli.py (hunk 1 above)
Comment (TahlonBrahic) [reply src/nix_auto_follow/cli.py:71]: The comment could be removed but it does let cases such as the below pass:
`    "flake-schemas": {
      "locked": {
        "lastModified": 1721999734,
        "narHash": "sha256-G5CxYeJVm4lcEtaO87LKzOsVnWeTcHGKbKxNamNWgOw=",
        "rev": "0a5c42297d870156d9c57d8f99e476b738dcd982",
        "revCount": 75,
        "type": "tarball",
        "url": "https://api.flakehub.com/f/pinned/DeterminateSystems/flake-schemas/0.1.5/0190ef2f-61e0-794b-ba14-e82f225e55e6/source.tar.gz"
      },
      "original": {
        "type": "tarball",
        "url": "https://flakehub.com/f/DeterminateSystems/flake-schemas/%3D0.1.5.tar.gz"
      }
    },`

Although we may need to remove it. What do you think?

File: src/nix_auto_follow/cli.py (hunk 2 above)
Comment (TahlonBrahic) [reply src/nix_auto_follow/cli.py:67]: I'll add a test file for sourcehut. Thank you!

File: src/nix_auto_follow/cli.py (hunk 1 above)
Comment (TahlonBrahic) [reply src/nix_auto_follow/cli.py:71]: I can go add a test case for the open pull request for the tarball. That should resolve this tarball issues for DeterminateSystems.

File: tests/test_cli.py [hunk 3]
```diff
@@ -115,6 +115,19 @@
...
             "git+ssh://git@gitlab.com/akibahmed/sops-secrects.git?ref=main&shallow=1",
         ),
+        (
+            Node.from_dict(
+                {
+                    "original": {
+                        "type": "sourcehut",
+                        "owner": "~rycee",
+                        "repo": "nmd",
+                        "url": "sourcehut:~rycee/nmd",
```
Comment (fzakaria) [inline tests/test_cli.py:125]: doesn't look like it has URL when i searched github for examples
```
      },
      "original": {
        "owner": "~tomterl",
        "repo": "fullframe",
        "type": "sourcehut"
      }
```
https://github.com/emacs-twist/examples/blob/a6e840259b545cc4083def36665b8e30bc781827/profiles/terlar/lock/flake.lock#L1391

Comment (fzakaria) [inline tests/test_cli.py:125]:
```suggestion
```

**Review by fzakaria — APPROVED** [review]

File: src/nix_auto_follow/cli.py (hunk 1 above)
Comment (fzakaria) [reply src/nix_auto_follow/cli.py]: thanks
